Preprocessing/tst_data.py

```diff
@@ -0,0 +1,1357 @@
+        for measures in self.measures_list:
+            self.logger.info(f"Read measures {measures['raw_fp']}")
+            with self.logger.indent:
+                measures["df"] = pd.read_csv(measures["raw_fp"], low_memory=False)
+
+    def _cleanup_measures(self):
+        """
+        Cleanup measures
+        + small typos fixed
+        + only expected columns
+        """
+        for measures in self.measures_list:
+            self.logger.info(f"Cleanup measures {measures['raw_fp']}")
+            with self.logger.indent:
+
+                # Strip spaces and \n on column names
+                measures["df"].columns = measures["df"].columns.str.strip(" \n")
+                EXPECTED_COLUMNS = (
+                    "Machine_Time",
+                    "Machine_N_cycles",
+                    "Machine_Displacement",
+                    "Machine_Load",
+                    r"MD_index--\d",
+                    r"MD_N_cycles--\d",
+                    r"MD_Displacement--\d",
+                    r"MD_Load--\d",
+                    r"u--\d",
+                    r"v--\d",
+                    r"exx--\d",
+                    r"eyy--\d",
+                    r"exy--\d",
+                    "Crack_length",
+                    "Crack_N_cycles",
+                    "Crack_Displacement",
+                    "Crack_Load",
+                    "Th_time",
+                    "Th_N_cycles",
+                    "Th_specimen_max",
+                    "Th_specimen_mean",
+                    "Th_chamber",
+                    "Th_uppergrips",
+                    "Th_lowergrips",
+                    r"T--\d",
+                    "Storage_modulus",
+                    "Tan_delta",
+                    "Specimen_name",
+                )
+
+                found_matching_columns = set()
+                for pattern in EXPECTED_COLUMNS:
+                    found_matching_columns.update(
+                        list(
+                            Experiment.__grep_matching_columns(
+                                pattern, measures["df"].columns
+                            )
+                        )
+                    )
+
+                # drop unexpected columns
+                unexpected_columns = list(
+                    set(measures["df"].columns) - found_matching_columns
+                )
+
+                if len(unexpected_columns) != 0:
+                    for unexpected_column in unexpected_columns:
+                        self.logger.warning(
+                            f"drop unexpected column {unexpected_column}"
+                        )
+                    measures["df"] = measures["df"].drop(columns=unexpected_columns)
+
+    def _validate_measures(self):
+        """
+        Validate measures
+        """
+        for measures in self.measures_list:
+            self.logger.info(f"Validate measures {measures['raw_fp']}")
+            with self.logger.indent:
+                EXPECTED_COLUMNS = {
```

Comment:
I wonder if we should move the `col_constraints` and `EXPECTED_COLS` variables as constants in the beginning of the file - since these data formats/constraints might evolve/get updated, but the logic of the functions usig them will not .. @sbancal

Reply:
Yes-n-No ...
I thought about it ... for sure it would be more readable and maintainable !
That meant more time for that ... and there are some fields that have a dynamic value (which will be generated depending on the data read from the file). See https://github.com/EPFL-ENAC/CCFatiguePlatform/blob/d0d9ad4da51fcaf038342bb6f53895e096b4acc6/Preprocessing/tst_data.py#L380

We'd need to get a workaround for this.